generators/python/src/fern_python/generators/sdk/client_generator/request_body_parameters/file_upload_request_body_parameters.py

```diff
@@ -124,13 +124,65 @@ def write(writer: AST.NodeWriter) -> None:
                     if property_as_union.type == "bodyProperty" and property_as_union.content_type is not None:
                         continue
                     elif property_as_union.type == "bodyProperty":
-                        writer.write_line(
-                            f'"{property_as_union.name.wire_value}": {self._get_body_property_name(property_as_union)},'
-                        )
+                        prop_name = self._get_body_property_name(property_as_union)
+                        # For multipart form data, httpx expects primitive types (str, int, etc.)
+                        # Object types need to be JSON serialized as strings
+                        if self._is_primitive_type(property_as_union.value_type):
+                            writer.write_line(f'"{property_as_union.name.wire_value}": {prop_name},')
+                        else:
+                            # JSON serialize complex types for multipart form compatibility
+                            writer.write(f'"{property_as_union.name.wire_value}": ')
+                            writer.write_node(
+                                AST.Expression(
+                                    Json.dumps(
+                                        AST.Expression(
+                                            self._context.core_utilities.jsonable_encoder(AST.Expression(prop_name))
+                                        )
+                                    )
+                                )
+                            )
+                            writer.write_line(",")
             writer.write_line("}")
 
         return AST.Expression(AST.CodeWriter(write))
 
+    def _is_primitive_type(self, type_reference: ir_types.TypeReference) -> bool:
```

Comment:
would be great to use a utility for this (I would think we already check this elsewhere?)